visualization/app/codeCharta/ui/mapTreeView/mapTreeView.component.spec.ts

```diff
@@ -107,5 +108,26 @@ describe("MapTreeViewController", () => {
 
 			expect(mapTreeViewController["_viewModel"].rootNode).toEqual(VALID_NODE_WITH_MULTIPLE_FOLDERS_SORTED_BY_NAME)
 		})
+
+		it("should clone new map before updating viewModel.rootNode to prevent unwanted changes in original map", () => {
+			const testNode = VALID_NODE_WITH_MULTIPLE_FOLDERS
+			const untouchedOriginalTestNode = klona(testNode)
+
+			mapTreeViewController["_viewModel"].rootNode = null
+
+			mapTreeViewController.onRenderMapChanged(testNode)
+			$timeout.flush(100)
```

Comment:
Could you please explain me what flush does and why you have to use it here?

Reply:
There are active timeouts in the code. $timeout.flush() flushes all pending timeouts, so that the process executes the callback function of the timeout immediately (if any).

Reply:
https://docs.angularjs.org/api/ngMock/service/$timeout#flush